analyzers/ruby/ruby.go

```diff
@@ -127,47 +127,86 @@ func (a *Analyzer) IsBuilt() (bool, error) {
 }
 
 func (a *Analyzer) Analyze() (graph.Deps, error) {
-	lockfilePath := filepath.Join(a.Module.Dir, "Gemfile.lock")
-	if a.Options.LockfilePath != "" {
-		lockfilePath = a.Options.LockfilePath
+	shouldFallback := true
```

Comment:
It would be easier to remove the fallback logic from the strategy helpers, and to instead only do fallback on the `default` case of the switch in this function.